packages/core/.gitignore

```diff
@@ -0,0 +1,6 @@
+# generated code
+src/**/*.js
```

Comment:
In retrospect (from raincatcher) I'd rather output the js to `dist/` and have a watcher->recompile command. No need for tacking this now of course.

Reply:
Purpose of this PR is mostly to tackcle infrastructure/scripts/build for typecripts.
More business specific code will be added in other PR.

I got impression that dist will make unit testing and publishing more complex.

Reply:
@wtrocki yes, I think there are always tradeoffs.
For using TS+lots of lerna modules we decided to output the compiled files to the same directories and include TS sources in the final npm packages instead of generated typings. This speeds up SDK development due to faster navigation and no need to constantly recompile internal dependencies.

However this also means that TS clients were trying compile from TS sources when getting the packages from npm, and that led to confusion sometimes.

An alternative is setting up file watchers for the packages and recompiling to js in a different directory like `dist/` and having `package.json#main = "./dist/index.js"`

It's a tradeoff between convenience during SDK development vs having the dev environment closer to the final published package shape... I don't think we need to worry about this right now though.